### PR TITLE
du -h: support GB

### PR DESCRIPTION
**Pull request by bootandy**
Allow -h (human readable form) to show GB (mimics original du behaviour)

### CODE REVIEW

File: src/du/du.rs [hunk 1]
```diff
@@ -290,7 +295,9 @@ pub fn uumain(args: Vec<String>) -> i32 {
 
     let convert_size = |size: u64| -> String {
         if matches.opt_present("human-readable") || matches.opt_present("si") {
-            if size >= mb {
+            if size >= gb {
```
Comment (Arcterus) [inline src/du/du.rs]: Shouldn't this support TB, etc. as well?

Comment (bootandy) [reply src/du/du.rs]: yeah it should, but then there will be 4 nearly duplicate lines not 3 :-(. I'll look at refactoring this bit.

File: src/du/du.rs [hunk 2]
```diff
@@ -289,18 +278,25 @@ pub fn uumain(args: Vec<String>) -> i32 {
     };
 
     let convert_size = |size: u64| -> String {
+        let multiplier: u64 = if matches.opt_present("si") {
+            1000
+        } else {
+            1024
+        };
+        let units = vec![("T", 4), ("G", 3), ("M", 2), ("K", 1)];
```
Comment (Arcterus) [inline src/du/du.rs]: Do you mind adding support for `E` and `P` as well?  `Y` and `Z` should also be marked as `TODO` or something as they are larger than a `u64` can handle.

Comment (bootandy) [reply src/du/du.rs]: Can do. 

Going to be offline a lot until easter so these changes are going to be slower.

File: src/du/du.rs [hunk 3]
```diff
@@ -289,18 +278,26 @@ pub fn uumain(args: Vec<String>) -> i32 {
     };
 
     let convert_size = |size: u64| -> String {
+        let multiplier: u64 = if matches.opt_present("si") {
+            1000
+        } else {
+            1024
+        };
+        // TODO: Suport Z & Y (currently limited by size of u64)
+        let units = vec![("E", 6), ("P", 5), ("T", 4), ("G", 3), ("M", 2), ("K", 1)];
```
Comment (Arcterus) [inline src/du/du.rs]: Another quick thing.  This could just be a static array instead of a dynamic one.

File: src/du/du.rs [hunk 4]
```diff
@@ -289,18 +278,26 @@ pub fn uumain(args: Vec<String>) -> i32 {
...
+
         if matches.opt_present("human-readable") || matches.opt_present("si") {
-            if size >= mb {
-                format!("{:.1}M", (size as f64) / (mb as f64))
-            } else if size >= kb {
-                format!("{:.1}K", (size as f64) / (kb as f64))
-            } else {
-                format!("{}B", size)
+            for &(unit, power) in units.iter() {
```
Comment (Arcterus) [inline src/du/du.rs]: Also it's probably more idiomatic to do `&units` as opposed to `units.iter()`.

File: src/du/du.rs (hunk 2 above)
Comment (Arcterus) [reply src/du/du.rs]: That's totally fine, take your time. :)

**Comment by bootandy** [comment]
@Arcterus code updated & squashed.

**Comment by Arcterus** [comment]
Thanks :)

**Pull request merged**